etna/transforms/detrend.py

```diff
@@ -15,7 +16,7 @@ class _OneSegmentLinearTrendBaseTransform(Transform):
     is_categorical = False
     is_static = False
 
-    def __init__(self, in_column: str, regressor: Optional[RegressorMixin] = None):
+    def __init__(self, in_column: str, regressor: Union[LinearModel, RegressorMixin] = None):
```

Comment:
There should be `Optional[Union[LinearModel, RegressorMixin]]` but `reggressor` is not optional in that case and it can't be equal to `None` ?

Reply:
Can it be that there are no examples in the code with `regressor=None`?

Reply:
Due to `strict_optional = False` parameter, Mypy resolves these annotation allowing None be true for every type annotation. 
In this case `None` can be safely removed.